packages/mds-transaction-service/service/validators.ts

```diff
@@ -14,56 +14,152 @@
+      amount: { type: 'integer' },
+      receipt: receiptSchema
+    },
+    required: ['amount', 'device_id', 'fee_type', 'provider_id', 'receipt', 'timestamp', 'transaction_id']
+  },
+  { keywords: ['example'] }
 )
 
 export const {
   validate: validateTransactionOperationDomainModel,
-  isValid: isValidTransactionOperationDomainModel
-} = schemaValidator<TransactionOperationDomainModel>(
-  Joi.object<TransactionOperationDomainModel>().keys({
-    transaction_id: Joi.string().uuid().required(),
-    operation_id: Joi.string().uuid().required(),
-    timestamp: Joi.number().integer().required(),
-    operation_type: Joi.string().required(),
-    author: Joi.string().required()
-  })
-)
+  $schema: TransactionOperationSchema
+} = SchemaValidator<TransactionOperationDomainModel>({
+  $id: 'TransactionOperation',
+  type: 'object',
+  properties: {
+    transaction_id: uuidSchema,
+    operation_id: uuidSchema,
+    timestamp: timestampSchema,
+    operation_type: { type: 'string', enum: [...TRANSACTION_OPERATION_TYPE] },
+    author: { description: 'Who/what executed this operation?', type: 'string' }
+  },
+  required: ['author', 'operation_id', 'operation_type', 'timestamp', 'transaction_id']
+})
 
 export const {
   validate: validateTransactionStatusDomainModel,
-  isValid: isValidTransactionStatusDomainModel
-} = schemaValidator<TransactionStatusDomainModel>(
-  Joi.object<TransactionStatusDomainModel>().keys({
-    transaction_id: Joi.string().uuid().required(),
-    status_id: Joi.string().uuid().required(),
-    timestamp: Joi.number().integer().required(),
-    status_type: Joi.string().required(),
-    author: Joi.string().required()
-  })
-)
+  $schema: TransactionStatusSchema
+} = SchemaValidator<TransactionStatusDomainModel>({
+  $id: 'TransactionStatus',
+  type: 'object',
+  properties: {
+    transaction_id: uuidSchema,
+    status_id: uuidSchema,
+    timestamp: timestampSchema,
+    status_type: { type: 'string', enum: [...TRANSACTION_STATUS_TYPE] },
+    author: { description: 'Who/what updated the status of the transaction?', type: 'string' }
+  },
+  required: ['author', 'status_id', 'status_type', 'timestamp', 'transaction_id']
+})
 
-export const { validate: validateTransactionId, isValid: isValidTransactionId } = schemaValidator<UUID>(
-  Joi.string().uuid().required()
-)
+export const { validate: validateTransactionId } = SchemaValidator<UUID>(uuidSchema)
 
-export const { validate: validateTransactionIds, isValid: isValidTransactionIds } = schemaValidator<UUID[]>(
-  Joi.array().items(Joi.string().uuid()).max(100).required()
-)
+export const { validate: validateTransactionIds } = SchemaValidator<UUID[]>({
+  type: 'array',
+  maxItems: 100,
+  items: uuidSchema
+})
+
+/* eslint-reason avoiding TS2742, TS4023, TS7056 */
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export const schemas: any[] = [TransactionSchema, TransactionOperationSchema, TransactionStatusSchema]
```

Comment:
I assume just allowing tuple inference causes issues? Will `AnySchema[]` work?

Reply:
I ended up going with `SchemaObject` which seems to do the trick, fixed in 624608c